jenkins/bisect.jpl

```diff
@@ -390,16 +390,34 @@ def findMergeBase(kdir, good, bad) {
             script: "git merge-base --is-ancestor ${base} HEAD")
 
         if (good_base != 0) {
-            def ref = "${params.REF_KERNEL_TREE}/${params.REF_KERNEL_BRANCH}"
+            def ref_url = params.REF_KERNEL_URL
+            def ref_tree = params.REF_KERNEL_TREE
+            def ref_branch = params.REF_KERNEL_BRANCH
+
+            if (!(ref_url && ref_tree && ref_branch)) {
+                dir(kci_core) {
+                    ref = sh(script: """\
+                                    ./kci_build get_reference --tree-name ${params.KERNEL_TREE} \
+                                    --branch ${params.KERNEL_BRANCH}""", returnStdout: true).trim().tokenize("\n")
+
+                    if (ref.size() > 0) {
```

Comment:
So, what happens if there's nothing in `params.REF_KERNEL_*` and `kci_build get_reference` fails and returns nothing?  I think we should return an error in this case and abort the job.

Reply:
I tried it several times and my observations are that the job exits with an error whenever the `sh` function gets a status code different than 0 and this is the case when kci_build returns nothing.
Nonetheless, I can make it more explicit with throwing `hudson.AbortException`

Reply:
OK as kci_build returns an error if the reference branch is not found, this will already abort the job so it's all fine.